### PR TITLE
Fixed spelling issues.

### PR DESCRIPTION
**Pull request by clintaire**
This pull request addresses several spelling and wording corrections in various files across the mypy repository. These changes are aimed at improving the readability of comments and documentation, ensuring clarity for future developers and contributors.

Changes Made:
№ 1: Corrected spelling errors such as “overriden” to “overridden” and “yields or yield froms” to “yields or yield forms”.
№ 2: Updated in-line comments to enhance their clarity while keeping the original meaning intact.
№ 3: Modified wording in CHANGELOG.md and other files to maintain consistency and precision.

↝ These corrections are purely to improve code readability and documentation quality. No functional changes were made to the codebase.

### CODE REVIEW

File: mypy/fastparse.py [hunk 1]
```diff
@@ -2249,7 +2249,7 @@ def visit_member_expr(self, e: MemberExpr) -> None:
 
 
 class FindYield(TraverserVisitor):
-    """Check if an AST contains yields or yield froms."""
+    """Check if an AST contains yields or yield forms."""
```
Comment (svisser) [inline mypy/fastparse.py:2252]: I think this refers to "yield from" (https://docs.python.org/3/whatsnew/3.3.html#pep-380), so it's written as intended?

File: mypyc/ir/module_ir.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def __init__(
         self.functions = functions
         self.classes = classes
         self.final_names = final_names
-        # Names of C statics used for Python 3.12 type variable objects.
+        # Names of C statistics used for Python 3.12 type variable objects.
```
Comment (brianschubert) [inline mypyc/ir/module_ir.py:31]: I think this refers to "C statics" as in "C static variables". That terminology is used elsewhere in mypyc ([example](https://github.com/python/mypy/blob/5c38427a53f409bae068c314e119fecc77730d76/mypyc/ir/ops.py#L796-L804)), so I think this is written as intended

File: mypy/fastparse.py (hunk 1 above)
Comment (clintaire) [reply mypy/fastparse.py:2252]: It seems like "from" is the correct terminology as per PEP 380 (https://docs.python.org/3/whatsnew/3.3.html#pep-380), so the original phrasing is accurate. The update might not be necessary.

File: mypyc/ir/module_ir.py (hunk 2 above)
Comment (clintaire) [reply mypyc/ir/module_ir.py:31]: Thanks for pointing that out! I'll leave it as is since "statics" seems to be the correct terminology.

**Pull request closed without merging**